test/hotspot/jtreg/compiler/loopopts/superword/TestMultiversionRemoveUselessSlowLoop.java

```diff
@@ -23,19 +23,72 @@
 
 package compiler.loopopts.superword;
 
+import compiler.lib.ir_framework.*;
+
 /*
  * @test
  * @bug 8350756
  * @summary Test case where the multiversion fast_loop disappears, and we should
  *          constant fold the multiversion_if, to remove the slow_loop.
  * @library /test/lib /
  * @run driver compiler.loopopts.superword.TestMultiversionRemoveUselessSlowLoop
- * @run main/othervm -XX:CompileCommand=compileonly,*Test*::test
- *                   -XX:-TieredCompilation -Xcomp -XX:PerMethodTrapLimit=0
- *                   compiler.loopopts.superword.TestMultiversionRemoveUselessSlowLoop
  */
 
 public class TestMultiversionRemoveUselessSlowLoop {
+
+    public static void main(String[] args) {
+        TestFramework framework = new TestFramework(TestMultiversionRemoveUselessSlowLoop.class);
+        // No traps means we cannot use the predicates version for SuperWord / AutoVectorization,
+        // and instead use multiversioning directly.
+        framework.addFlags("-XX:-TieredCompilation", "-XX:PerMethodTrapLimit=0");
+        framework.setDefaultWarmup(0); // simulates Xcomp
+        framework.start();
+    }
+
+    public static final int SIZE = 20;
+    public static final int[] a = new int[SIZE];
+    public static final int[] b = new int[SIZE];
+    public static final int SIZE2 = 10_000;
+    public static final int[] a2 = new int[SIZE2];
+    public static final int[] b2 = new int[SIZE2];
+
+    @Test
+    @IR(counts = {"pre .* multiversion_fast",  "= 2", // regular pre-main-post for both loops
+                  "main .* multiversion_fast", "= 2",
+                  "post .* multiversion_fast", "= 2",
+                  "multiversion_delayed_slow", "= 2", // both have the delayed slow_loop
+                  "multiversion",              "= 8"}, // nothing unexpected
```

Comment:
Should we also add a match on `IRNode.OPAQUE_MULTIVERSIONING`?

Reply:
Done :)